packages/electrode-archetype-react-app-dev/lib/dev-admin/admin-server.js

```diff
@@ -18,35 +20,32 @@ const PROXY_SERVER_NAME = "Electrode Dev Proxy";
 const DEV_PROXY_ENABLED = Boolean(process.env.APP_SERVER_PORT);
 
 class AdminServer {
-  constructor(args) {
+  constructor(args, options) {
     this._opts = args.opts;
     this._passThru = args._;
     this._messageId = 1;
     this._saveWebpackReportData = undefined;
+    this._webpackDevRelay = new WebpackDevRelay();
     this._servers = {};
+    this._io = (options && options.inputOutput) || new ConsoleIO();
   }
 
   async start() {
     this._wds = ck`<gray.inverse>[wds]</> `;
     this._proxy = ck`<green.inverse>[proxy]</> `;
-    this.setupConsoleInterface();
+    this._io.setup();
     this.handleUserInput();
-    await this.startDevServer();
+    await this.startWebpackDevServer();
     await this.startAppServer();
     if (DEV_PROXY_ENABLED) {
       // to debug dev proxy
       // await this.startProxyServer("--inspect-brk");
       await this.startProxyServer();
     }
-    setTimeout(() => this.showConsoleMenu(), 100);
+    setTimeout(() => this.showMenu(), 100);
```

Comment:
Why not `setTimeout(this.showMenu, 100);`?

Reply:
that would lose `this`.  It'd need to be `setTimeout(this.showMenu.bind(this), 100)`

Reply:
Typically I tend to avoid passing function directly into standard APIs like this, because you can't guarantee what params a function takes and what the API would pass to the function.
For example, `parseInt` takes two params and it's a thing where `["1", "2"].map(parseInt)` is not `[1, 2]`, because `parseInt` takes two params: `(str, radix)` and `map` pass two params: `(value, index)`